code/modules/reagents/chemistry/reagents/impure_reagents.dm

```diff
@@ -37,3 +37,26 @@
 	ph = 1.5
 	taste_description = "an awful, strongly chemical taste"
 	color = "#270d03"
+
+/////////Unique
+
+/datum/reagent/impurity/eigenswap
+	name = "Eigenswap"
+	description = "This reagent is known to swap the handedness of a patient."
+	ph = 3.3
+	chemical_flags = REAGENT_DONOTSPLIT
+
+/datum/reagent/impurity/eigenswap/on_mob_life(mob/living/carbon/carbon_mob)
+	. = ..()
+	if(prob(creation_purity*100))
+		var/list/cached_hand_items = carbon_mob.held_items
+		var/index = 1
+		for(var/thing in cached_hand_items)
+			index++
+			if(index > length(cached_hand_items))//If we're past the end of the list, go back to start
+				index = 1
+			if(!thing)
+				continue
+			carbon_mob.put_in_hand(thing, index, TRUE, TRUE)
+			playsound(carbon_mob, 'sound/effects/phasein.ogg', 20, TRUE)
```

Comment:
```suggestion
	if(!prob(creation_purity * 100))
		return
	var/list/cached_hand_items = carbon_mob.held_items
	var/index = 1
	for(var/thing in cached_hand_items)
		index++
		if(index > length(cached_hand_items))//If we're past the end of the list, go back to start
			index = 1
		if(!thing)
			continue
		carbon_mob.put_in_hand(thing, index, TRUE, TRUE)
		playsound(carbon_mob, 'sound/effects/phasein.ogg', 20, TRUE)
```
* Early return and spaces.
* How will this play with no-drop items such as armblades and zombie hands?

Reply:
That's a good question - I assume it'll just swap them around and they'll still work, but I'll test it!

Reply:
It teleports them back and forth, but I can't drop them still. So seems fine!